server/odc-service/src/main/java/com/oceanbase/odc/service/task/config/JobSecurityChecker.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.task.config;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Method marked by ${@link JobSecurityChecker} will be checked remote ip when task-framework is running in process mode.
+ * reference ${@link JobSecurityCheckerAspect}
+ * 
+ * @author yaobin
+ * @date 2024-03-04
+ * @since 4.2.4
+ */
+@Target(value = {ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface JobSecurityChecker {
```

Comment:
use Filter  may more flexialble , due:
- for desktop version, all API should limited to access from local only;
- also, there also may exists other API may be limited for Web version. not only for task framework.

Reply:
okey, 
1.add LocalRequestFilter limit access `TaskController` 
2.modify `DesktopSecurityFilter` limit all request should be localhost